src/utils/__mocks__/redis.ts

```diff
@@ -0,0 +1,23 @@
+import Redis from 'ioredis';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const RedisMock = require('ioredis-mock');
+
+const clients: Redis[] = [];
```

Comment:
Maybe making our own class for the redis client and operations (our own error handled `get` and `setEx` methods) would be good. Keep the `clients` array out of global memory.

Reply:
I see this is tho mocked one, carry comments over to unmocked.